clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java

```diff
@@ -167,21 +167,9 @@ public void ensureValid() {
      *         {@link RecordBatch#NO_TIMESTAMP} if the batch is empty
      */
     public long baseTimestamp() {
-        return buffer.getLong(FIRST_TIMESTAMP_OFFSET);
-    }
-
-    /**
-     * Get the timestamp of the first record in this batch. It is usually the create time of the record even if the
-     * timestamp type of the batch is log append time.
-     * 
-     * @return The first timestamp if a record has been appended, unless the delete horizon has been set
-     *         {@link RecordBatch#NO_TIMESTAMP} if the batch is empty or if the delete horizon is set
-     */
-    public long firstTimestamp() {
-        final long baseTimestamp = baseTimestamp();
         if (hasDeleteHorizonMs())
             return RecordBatch.NO_TIMESTAMP;
```

Comment:
Don't we want to remove this? We still need to be able to access the base timestamp in order to compute the successive timestamps in the record iterator, right?

Reply:
yes you are right. sorry for the mistake. It should be fixed now